xendit-java-lib/src/main/java/com/xendit/model/Customer.java

```diff
@@ -0,0 +1,147 @@
+   * @param description Merchant-provided description for the customer object.
+   * @param phoneNumber Alternate or landline phone number in E.164 international standard.
+   * @param nationality 2-letter ISO 3166-2 country code for the customer's nationality.
+   * @param addresses Array of objects containing the specific customer's address infromation.
+   * @param dateOfBirth Date of birth of the customer in YYYY-MM-DD format.
+   * @param metadata Object of additional information the user may use.
+   * @return Customer model.
+   * @throws XenditException XenditException
+   */
+  public static Customer createCustomer(
+      String referenceId,
+      String mobileNumber,
+      String email,
+      String givenNames,
+      String middleName,
+      String surname,
+      String description,
+      String phoneNumber,
+      String nationality,
+      CustomerAddress[] addresses,
+      String dateOfBirth,
+      Map<String, Object> metadata)
+      throws XenditException {
+    Map<String, Object> params = new HashMap<>();
+    params.put("reference_id", referenceId);
+    params.put("mobile_number", mobileNumber);
+    params.put("email", email);
+    params.put("given_names", givenNames);
+    params.put("middle_name", middleName);
+    params.put("surname", surname);
+    params.put("description", description);
+    params.put("phone_number", phoneNumber);
+    params.put("nationality", nationality);
+    params.put("addresses", addresses);
+    params.put("date_of_birth", dateOfBirth);
+    params.put("metadata", metadata);
+    return createCustomerRequest(new HashMap<>(), params);
+  }
+
+  /**
+   * Create customer with all parameter as HashMap
+   *
+   * @param params listed here https://developers.xendit.co/api-reference/#create-customer.
+   * @return Customer
+   * @throws XenditException
+   */
+  public static Customer createCustomer(Map<String, Object> params) throws XenditException {
+    return createCustomerRequest(new HashMap<>(), params);
+  }
+
+  /**
+   * Create customer with headers and all parameter as HashMap
+   *
+   * @param headers
+   * @param params listed here https://developers.xendit.co/api-reference/#create-customer.
+   * @return Customer
+   * @throws XenditException
+   */
+  public static Customer createCustomer(Map<String, String> headers, Map<String, Object> params)
+      throws XenditException {
+    return createCustomerRequest(headers, params);
+  }
+
+  /**
+   * Get customer by reference id
+   *
+   * @param referenceId Customer reference ID
+   * @return Customer
+   * @throws XenditException XenditException
+   */
+  public static Customer[] getCustomerByReferenceId(String referenceId) throws XenditException {
+    String url = String.format("%s%s%s", Xendit.getUrl(), "/customers?reference_id=", referenceId);
+    return Xendit.requestClient.request(RequestResource.Method.GET, url, null, Customer[].class);
+  }
+
+  public static Customer createCustomerRequest(
```

Comment:
Don't think we normally expose this as `public` but cmiiw

Reply:
ah yes I think it must be private